app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    user.posts_counter += 1
+  end
+
+  def recent_comments
+    comments.all.order(created_at: :desc).limit(5)
+  end
+end
```

Comment:
- [ ] [OPTIONAL] I would like to also suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

```ruby
after_save :update_post_counter

  private

  def update_post_counter
   Code here ...
  end
```

Reply:
**Thank you very much @Whoistolu I will add the recommended codes.**